### PR TITLE
Add Reclaim contract

### PR DESCRIPTION
**Pull request by Recmo**

### CODE REVIEW

File: contracts/Commitment/WhitelistedCommitment.sol [hunk 1]
```diff
@@ -1,8 +1,10 @@
 pragma solidity 0.4.15;
 
 import './PublicCommitment.sol';
+import '../Reclaimable.sol';
 
-contract WhitelistedCommitment is AccessControlled, AccessRoles, CommitmentBase {
+
+contract WhitelistedCommitment is AccessControlled, AccessRoles, CommitmentBase, Reclaimable {
```
Comment (rudolfix) [inline contracts/Commitment/WhitelistedCommitment.sol]: It should not allow to reclaim `Neumark` token before WhitelistedCommitment is finalized,  Neumarks are already handled differently
* `abortCommitment()` will burn all neumarks in possession of the contract if called before `startDate`
* no one should be able to reclaim or burn Neumarks when commitment is in progress

use `isFinalized()` function to get finalization status
All neumarks in contract possession after finalization can be reclaimed

File: contracts/Commitment/PublicCommitment.sol [hunk 2]
```diff
@@ -43,11 +44,14 @@ contract PublicCommitment is CommitmentBase {
     }
 
     function PublicCommitment(
+        IAccessPolicy _policy,
```
Comment (rudolfix) [inline contracts/Commitment/PublicCommitment.sol]: You may move it to `BaseCommitment` class in that case, it seems that `AccessControlled` and `Reclaimable` are obligatory for any commitment contract.

File: test/Reclaimable.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    expect(reclaimerAfter.sub(reclaimerBefore)).to.be.bignumber.equal(amount);
+  });
+  it("should only allow ROLE_RECLAIMER to reclaim", async () => {
+    const amount = web3.toWei(1, "ether");
+    await forceEther(reclaimable.address, amount);
+    await expectThrow(reclaimable.reclaim(RECLAIM_ETHER, { from: other }));
+    await expectThrow(reclaimable.reclaim(RECLAIM_ETHER, { from: deployer }));
+    await reclaimable.reclaim(RECLAIM_ETHER, { from: reclaimer });
+  });
+});
```
Comment (rudolfix) [inline test/Reclaimable.js]: I would add following pending tests, we should implement them later
```
it("should not reclaim Neumark token before WhitelistedCommitment is finalized");
it("should not be able to reclaim ether from EtherToken");
it("should not be able to reclaim assetToken from LockedAccount");
```

Comment (Recmo) [reply test/Reclaimable.js]: Added to the respective tests

File: contracts/Commitment/PublicCommitment.sol (hunk 2 above)
Comment (Recmo) [reply contracts/Commitment/PublicCommitment.sol]: Done.

**Pull request merged**